centreon/centreon-plugin.py

```diff
@@ -0,0 +1,134 @@
+    if not any(g['name'] == 'rudder-nodes' for g in centreon_hosts.gethostgroup(nodeid)['result']):
+        print("[ ] Rudder node " + nodeid + " not in Centreon host group 'rudder-nodes'. Adding it...")
+        centreon_hosts.addhostgroup(nodeid, ['rudder-nodes'])
+
+# When manually used, used for pulling
+def updateNodesJSON(confPath):
+    print("[ ] Pulling data from Rudder server API...")
+    data = getRequestToRudderAPI("/nodes")
+    rnodes = []
+    for node in data['data']['nodes']:
+        rnodes.append(dictifyNode(node))
+    nodes_file = open(confPath + 'rudder_nodes.json', 'w+')
+    json.dump(rnodes, nodes_file)
+    nodes_file.close()
+    print("[+] Done")
+
+# Manual pushing
+def updateCentreonHosts(confPath, poller):
+    print("[ ] Checking if Centreon is up-to-date...")
+    checkRudderCentreonHostGroup()
+    centreon_hosts = Host()
+    rudder_nodes = json.load(open(confPath + 'rudder_nodes.json'))
+    
+    for rn in rudder_nodes:
+        if not any(h['name'] == rn['rudder_id'] for h in centreon_hosts.list()['result']):
+            print("[ ] Unregistered Rudder node found (id " + rn['rudder_id'] + "). Adding it to Centreon...")
+            centreon_hosts.add(rn['rudder_id'], "Rudder " + rn['node_type'] + " node " + rn['hostname'], rn['ip_address'], '', poller, '')
+            checkIfNodeInRudderGroup(rn['rudder_id'], centreon_hosts)
+    for ch in centreon_hosts.list()['result']:
+        if any(g['name'] == 'rudder-nodes' for g in centreon_hosts.gethostgroup(ch['name'])['result']) and not any(ch['name'] == rn['rudder_id'] for rn in rudder_nodes):
+                print("[ ] Host " + ch['name'] + " not listed in Rudder but appearing in rudder-nodes Centreon host group. Deleting host...")
+                centreon_hosts.delete(ch['name'])
+                print("[+] Done")
+    print("[+] Done")
+
+def checkRudderCentreonHostGroup():
+    centreon_hostGrps = Hostgroups()
+    if not any(g['name'] == 'rudder-nodes' for g in centreon_hostGrps.list()['result']):
+        print("[ ] Host group 'rudder-nodes' not found on Centreon Central server, creating it...")
+        centreon_hostGrps.add("rudder-nodes", "Rudder nodes Hosts Group")
+
+def addHostOnRudderHook(rudderID, poller):
+    data = getRequestToRudderAPI("/nodes/" + rudderID)
+    node = dictifyNode(data['data']['nodes'][0])
+    centreon_hosts = Host()
+    if not any(h['name'] == node['rudder_id'] for h in centreon_hosts.list()['result']):
+        centreon_hosts.add(node['rudder_id'], "Rudder " + node['node_type'] + " node " + node['hostname'], node['ip_address'], '', poller, '')
+    checkIfNodeInRudderGroup(node['rudder_id'], centreon_hosts)
+
+def delHostOnRudderHook(rudderID):
+    centreon_hosts = Host()
+    if any(h['name'] == rudderID for h in centreon_hosts.list()['result']):
+        centreon_hosts.delete(rudderID)
+
+if __name__ == '__main__':
+    args = docopt(__doc__)
+    confPath = "/opt/rudder/etc/"
```

Comment:
This line should be at the top of the file, and it should just contain the full path of the configuration file